pkg/utils/workflow.go

```diff
@@ -0,0 +1,181 @@
+		if err != nil {
+			return err
+		}
+	} else if workflowKind == "CronWorkflow" {
+
+		var cronWorkflow v1alpha1.CronWorkflow
+		err = UnmarshalObject(body, &cronWorkflow)
+		if err != nil {
+			return err
+		}
+
+		// Marshal the workflow back to JSON for API payload.
+		workflowStr, _ := json.Marshal(cronWorkflow)
+		chaosWorkFlowInput.WorkflowManifest = string(workflowStr)
+		chaosWorkFlowInput.WorkflowName = cronWorkflow.ObjectMeta.Name
+		chaosWorkFlowInput.IsCustomWorkflow = true
+
+		// Set the schedule for the workflow
+		chaosWorkFlowInput.CronSyntax = cronWorkflow.Spec.Schedule
+
+		// Fetch the weightages for experiments present in the spec.
+		err = FetchWeightages(chaosWorkFlowInput, cronWorkflow.Spec.WorkflowSpec.Templates)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("Invalid resource kind found in manifest.")
+	}
+
+	return nil
+}
+
+// FetchWeightages takes in the templates present in the workflow spec and
+// assigns weightage to each of the experiments present in them. It can parse
+// both artifacts and remote experiment specs.
+func FetchWeightages(chaosWorkFlowInput *model.ChaosWorkFlowRequest, templates []v1alpha1.Template) error {
+
+	var chaosExperiments []chaosTypes.ChaosExperiment
+
+	// Fetch all present experiments and append them to the experiments array
+	for _, t := range templates {
+		var c chaosTypes.ChaosExperiment
+
+		// Only the template named "install-chaos-experiments" contains ChaosExperiment(s)
+		if t.Name == "install-chaos-experiments" {
+
+			if len(t.Inputs.Artifacts) != 0 {
+
+				// These are experiments with the spec passed as an artifact
+				for _, a := range t.Inputs.Artifacts {
+					err := yaml.Unmarshal([]byte(a.Raw.Data), &c)
+					if err != nil {
+						return errors.New("Error parsing ChaosExperiment: " + err.Error())
+					}
+					chaosExperiments = append(chaosExperiments, c)
+				}
+
+			} else { // These are experiments with their spec passed as remote file URLs
+
+				// Extracting the URL from the container arguments
+				re := regexp.MustCompile(`kubectl apply -f\s(?P<first>.+)\s-n.+`)
+				extractURL := fmt.Sprintf("${%s}", re.SubexpNames()[1])
+				fileURL := re.ReplaceAllString(t.Container.Args[0], extractURL)
+
+				body, err := ReadRemoteFile(fileURL)
+				if err != nil {
+					return errors.New("Error reading ChaosExperiment: " + err.Error())
+				}
+
+				// Using a decoder since there might be multiple YAML documents
+				// inside the same remote file.
+				decoder := yamlutil.NewYAMLToJSONDecoder(bytes.NewReader(body))
+				for {
+					if err := decoder.Decode(&c); err != nil {
+						break
+					}
+					chaosExperiments = append(chaosExperiments, c)
+				}
+			}
+		}
+	}
+
+	// Assign weights to each chaos experiment
+	for _, c := range chaosExperiments {
+
+		// Fetch experiment weightage from annotation
+		w, ok := c.ObjectMeta.Annotations["litmuschaos.io/experiment-weightage"]
```

Comment:
Instead of adding annotations, can we add labels to the ChaosEngine input artifacts? From the backend side, we are adding weights label to the specific input artifact. 
Attaching a workflow manifest for reference:
[custom-chaos-workflow.txt](https://github.com/litmuschaos/litmusctl/files/8725418/custom-chaos-workflow.txt)
Also the weight name will be ChaosEngine generateName, this change was added in 2.9.0 to fix the issues with multiple experiments with same name.

Reply:
> Instead of adding annotations, can we add labels to the ChaosEngine input artifacts?

I see. Let me know if this logic sounds good then,
```
- Fetch all artifact names in the `install-chaos-experiments` template.
- Iterate through the rest of the templates whose names are present in the list of names fetched after step 1 (essentially the templates containing the ChaosExperiments).
	- Store `t.metadata.labels.weight` as the weight.
```

Reply:
> Also the weight name will be ChaosEngine generateName, this change was added in 2.9.0 to fix the issues with multiple experiments with same name.

Noted. Switching the name from `c.ObjectMeta.Name` to `c.ObjectMeta.GenerateName`.

Reply:
Yes, this is correct. One minor change in https://github.com/litmuschaos/litmusctl/pull/64#discussion_r876636317
Instead of ChaosExperiments, its ChaosEngine
Apart from that, the logic looks good ✅ 
You can take a look here for a better understanding: https://github.com/litmuschaos/litmus/blob/5f9400c37a1663dc28b2424c3d25bb94f0ebeffe/litmus-portal/graphql-server/pkg/chaos-workflow/ops/operations.go#L328

Reply:
> Also the weight name will be ChaosEngine generateName, this change was added in 2.9.0 to fix the issues with multiple experiments with same name.

Please see this [commit](https://github.com/litmuschaos/litmusctl/pull/64/commits/8a501c11750dfae6f03f0a673eb7b359ae5e0f0f).

For the updated weight assignment logic, see this [commit](https://github.com/litmuschaos/litmusctl/pull/64/commits/eb6916395a60a26fa75c832d9d9bbb0ea95c8b1f).

Reply:
For updated weight assignment logic, see this [commit](https://github.com/litmuschaos/litmusctl/pull/64/commits/9a9c21d8f91435f1f19e354cca4c3d0c4bb34981). For the change in how weight name is fetched, see this [commit](https://github.com/litmuschaos/litmusctl/pull/64/commits/ae5d2821219bb30175ad8773c11891698fa40b22).